core/src/main/java/io/grpc/internal/ManagedChannelImpl2.java

```diff
@@ -0,0 +1,874 @@
+          "scheduledExecutor is already cleared. Looks like you are calling this method after "
+          + "you've already shut down");
+      final OobChannel oobChannel = new OobChannel(censusFactory, authority, executor,
+          scheduledExecutorCopy, stopwatchSupplier, channelExecutor);
+      final InternalSubchannel internalSubchannel = new InternalSubchannel(
+          addressGroup, authority, userAgent, backoffPolicyProvider, transportFactory,
+          scheduledExecutorCopy, stopwatchSupplier, channelExecutor,
+          // All callback methods are run from channelExecutor
+          new InternalSubchannel.Callback() {
+            @Override
+            void onTerminated(InternalSubchannel is) {
+              oobChannels.remove(is);
+              oobChannel.handleSubchannelTerminated();
+              maybeTerminateChannel();
+            }
+
+            @Override
+            void onStateChange(InternalSubchannel is, ConnectivityStateInfo newState) {
+              oobChannel.handleSubchannelStateChange(newState);
+            }
+          });
+      oobChannel.setSubchannel(internalSubchannel);
+      channelExecutor.executeLater(new Runnable() {
+          @Override
+          public void run() {
+            if (terminating) {
+              oobChannel.shutdown();
+            }
+            if (!terminated) {
+              // If channel has not terminated, it will track the subchannel and block termination
+              // for it.
+              oobChannels.add(internalSubchannel);
+            }
+          }
+        }).drain();
+      return oobChannel;
+    }
+
+    @Override
+    public String getAuthority() {
+      return ManagedChannelImpl2.this.authority();
+    }
+
+    @Override
+    public NameResolver.Factory getNameResolverFactory() {
+      return nameResolverFactory;
+    }
+
+    @Override
+    public void runSerialized(Runnable task) {
+      channelExecutor.executeLater(task).drain();
+    }
+
+    @Override
+    public void updatePicker(final SubchannelPicker picker) {
+      runSerialized(new Runnable() {
+          @Override
+          public void run() {
+            subchannelPicker = picker;
+            delayedTransport.reprocess(picker);
+          }
+        });
+    }
+  }
+
+  @Override
+  public LogId getLogId() {
+    return logId;
+  }
+
+  private class NameResolverListenerImpl implements NameResolver.Listener {
+    final LoadBalancer2 balancer;
+
+    NameResolverListenerImpl(LoadBalancer2 balancer) {
+      this.balancer = balancer;
+    }
+
+    @Override
+    public void onUpdate(final List<ResolvedServerInfoGroup> servers, final Attributes config) {
+      if (servers.isEmpty()) {
+        onError(Status.UNAVAILABLE.withDescription("NameResolver returned an empty list"));
+        return;
+      }
+      log.log(Level.FINE, "[{0}] resolved address: {1}, config={2}",
+          new Object[] {getLogId(), servers, config});
+      channelExecutor.executeLater(new Runnable() {
```

Comment:
Why isn't the helper accessible here? It looks like it is to me. Maybe the code changed and now it is.